dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -96,83 +96,74 @@ def _batched_eigh(a, UPLO, eigen_mode, w_type, v_type):
 
     """
 
-    is_cpu_device = a.sycl_device.has_aspect_cpu
-    orig_shape = a.shape
-    # get 3d input array by reshape
-    a = a.reshape(-1, orig_shape[-2], orig_shape[-1])
-    a_usm_arr = dpnp.get_usm_ndarray(a)
-
-    # allocate a memory for dpnp array of eigenvalues
-    w = dpnp.empty_like(
-        a,
-        shape=orig_shape[:-1],
-        dtype=w_type,
-    )
-    w_orig_shape = w.shape
-    # get 2d dpnp array with eigenvalues by reshape
-    w = w.reshape(-1, w_orig_shape[-1])
-
     # `eigen_mode` can be either "N" or "V", specifying the computation mode
     # for OneMKL LAPACK `syevd` and `heevd` routines.
     # "V" (default) means both eigenvectors and eigenvalues will be calculated
     # "N" means only eigenvalues will be calculated
     jobz = _jobz[eigen_mode]
     uplo = _upper_lower[UPLO]
 
-    # Get LAPACK function (_syevd for real or _heevd for complex data types)
+    # Get LAPACK function (_syevd_batch for real or _heevd_batch
+    # for complex data types)
     # to compute all eigenvalues and, optionally, all eigenvectors
     lapack_func = (
-        "_heevd" if dpnp.issubdtype(v_type, dpnp.complexfloating) else "_syevd"
+        "_heevd_batch"
+        if dpnp.issubdtype(v_type, dpnp.complexfloating)
+        else "_syevd_batch"
     )
 
     a_sycl_queue = a.sycl_queue
-    a_order = "C" if a.flags.c_contiguous else "F"
 
-    # need to loop over the 1st dimension to get eigenvalues and
-    # eigenvectors of 3d matrix A
-    batch_size = a.shape[0]
-    eig_vecs = [None] * batch_size
-    ht_list_ev = [None] * batch_size * 2
-    for i in range(batch_size):
-        # oneMKL LAPACK assumes fortran-like array as input, so
-        # allocate a memory with 'F' order for dpnp array of eigenvectors
-        eig_vecs[i] = dpnp.empty_like(a[i], order="F", dtype=v_type)
+    a_orig_shape = a.shape
+    # get 3d input array by reshape
+    a = a.reshape(-1, a_orig_shape[-2], a_orig_shape[-1])
```

Comment:
```suggestion
    a = dpnp.reshape(a, (-1, a_orig_shape[-2], a_orig_shape[-1]))
```
Otherwise it fails for inputs given as `usm_ndarray`.

Reply:
Thank you for noticing it!
These changes also require a new PR for all linear functions.